src/utils/mcpIntegration.ts

```diff
@@ -0,0 +1,186 @@
+import { experimental_createMCPClient } from 'ai';
+import { StreamableHTTPClientTransport } from '@modelcontextprotocol/sdk/client/streamableHttp.js';
+
+export interface MCPServer {
+  id: string;
+  name: string;
+  endpoint: string;
+  description?: string;
+  enabled: boolean;
+  apiKey?: string;
+  authMethod?: 'url' | 'header';
+}
+
+export interface MCPClientManager {
+  clients: Map<string, any>;
+  tools: any;
+}
+
+export class MCPIntegration {
+  private clientManager: MCPClientManager;
+  private servers: MCPServer[];
+
+  constructor() {
+    this.clientManager = {
+      clients: new Map(),
+      tools: {},
+    };
+    this.servers = [];
+  }
+
+  async initializeServers(servers: MCPServer[]): Promise<void> {
+    this.servers = servers.filter(server => server.enabled);
+
+    await this.cleanup();
```

Comment:
Instead of cleaning up all the servers every time we call initializeServers, is it possible to to read the `clients` map and add MCP servers that are missing from `servers`, and remove MCP servers from `clients` map if they no longer exist in `servers`?